zim/tmp-kiwix/cleanup-old-files.py

```diff
@@ -0,0 +1,120 @@
+
+def list_files_to_delete(folder_path, days):
+    """
+    List files that are older than a specified number of days.
+
+    Args:
+        folder_path (str): The path to the target folder.
+        days (int): Delete files older than this many days.
+
+    Returns:
+        list: A list of file paths to be deleted.
+    """
+    current_time = datetime.datetime.now()
+    files_to_delete = []
+
+    def list_files(folder_path):
+        for foldername, subfolders, filenames in os.walk(folder_path):
+            for filename in filenames:
+                file_path = os.path.join(foldername, filename)
+                yield file_path
+
+    for file_path in list_files(folder_path):
+        # Get the last modification time of the file
+        file_time = datetime.datetime.fromtimestamp(os.path.getmtime(file_path))
+
+        # Calculate the difference in days
+        days_difference = (current_time - file_time).days
+
+        if days_difference > days:
+            files_to_delete.append(file_path)
+
+    return files_to_delete
+
+def list_directories_to_delete(folder_path, files_to_delete):
+    """
+    List empty subdirectories or subdirectories containing only files to be deleted.
+
+    Args:
+        folder_path (str): The path to the target folder.
+        files_to_delete (list): A list of file paths to be deleted.
+
+    Returns:
+        list: A list of empty subdirectories to be deleted.
+    """
+    empty_directories_to_delete = []
+
+    for foldername, subfolders, filenames in os.walk(folder_path):
+        for subfolder in subfolders:
+            subfolder_path = os.path.join(foldername, subfolder)
+            
+            # Check if the directory is already empty or contains only files that will be deleted
+            if not os.listdir(subfolder_path) or all(os.path.join(subfolder_path, filename) in files_to_delete for filename in os.listdir(subfolder_path)):
+                empty_directories_to_delete.append(subfolder_path)
+
+    return empty_directories_to_delete
+
+def process_deletion(dry_run, files_to_delete, empty_directories_to_delete):
+    """
+    Process file and directory deletion based on the dry-run status.
+
+    Args:
+        dry_run (bool): True for dry-run, False for actual deletion.
+        files_to_delete (list): A list of file paths to be deleted.
+        empty_directories_to_delete (list): A list of empty subdirectories to be deleted.
+    """
+    if dry_run:
+        print(f"These files would be deleted:")
+        print("\n".join(files_to_delete))
+        print("\nEmpty subdirectories that would be deleted:")
+        print("\n".join(empty_directories_to_delete))
+    else:
+        print(f"Deleting files:")
+        for file_path in files_to_delete:
+            os.remove(file_path)
+            print(f"Deleted: {file_path}")
+
+        print(f"\nDeleting empty subdirectories:")
+        for directory_path in empty_directories_to_delete:
+            os.rmdir(directory_path)
```

Comment:
You may want to catch your `Path().rmdir()` in case it's not empty for some reason (might be that someone else pushed something in) and not just crash the script